src/main/java/org/opensearch/neuralsearch/processor/normalization/MinMaxScoreNormalizationTechnique.java

```diff
@@ -34,8 +42,28 @@
 public class MinMaxScoreNormalizationTechnique implements ScoreNormalizationTechnique, ExplainableTechnique {
```

Comment:
We have added static class LowerBound in the same class. Can we extract it to independent file? If in future we add similar functionality then we have to rework to extract Lower bound and put it in another file.

Reply:
not sure I got your reasoning. I don't think it needs to be in a separate file, it's essentially a record class, and I don't see a need for it in other techniques in near future. We can do this later if needed, currently it's over engineering